contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -123,42 +142,54 @@ contract MixinSignatureValidator is
         );
     }
 
-    /// @dev Verifies that a hash has been signed by the given signer.
-    /// @param hash Any 32-byte hash.
-    /// @param signerAddress Address that should have signed the given hash.
-    /// @param signature Proof that the hash has been signed by signer.
-    /// @return True if the signature is valid for the given hash and signer.
-    function isValidHashSignature(
-        bytes32 hash,
+    /// @dev Verifies that a signature for a transaction is valid.
+    /// @param transaction The transaction.
+    /// @param signerAddress Address that should have signed the given order.
+    /// @param signature Proof that the order has been signed by signer.
+    /// @return isValid `true` if the signature is valid for the given transaction and signer.
+    function isValidTransactionSignature(
+        ZeroExTransaction memory transaction,
         address signerAddress,
         bytes memory signature
     )
         public
         view
         returns (bool isValid)
     {
-        SignatureType signatureType = _readValidSignatureType(
-            hash,
+        bytes32 transactionHash = getTransactionHash(transaction);
+        return _isValidTransactionWithHashSignature(
+            transaction,
+            transactionHash,
             signerAddress,
             signature
         );
-        // Only hash-compatible signature types can be handled by this
-        // function.
-        if (signatureType == SignatureType.OrderValidator ||
-            signatureType == SignatureType.WalletOrderValidator) {
-            _rrevert(SignatureError(
-                SignatureErrorCodes.INAPPROPRIATE_SIGNATURE_TYPE,
-                hash,
-                signerAddress,
-                signature
-            ));
-        }
-        return _validateHashSignatureTypes(
-            signatureType,
+    }
+
+    /// @dev Checks if a signature is of a type that should be verified for
+    /// every action.
+    /// @param hash The hash of the order/transaction.
+    /// @param signerAddress The address of the signer.
+    /// @param signature The signature for `hash`.
+    /// @return needsRegularValidation True if the signature should be validated
+    ///                                for every action.
+    function doesSignatureRequireRegularValidation(
+        bytes32 hash,
+        address signerAddress,
+        bytes memory signature
+    )
+        public
+        pure
+        returns (bool needsRegularValidation)
```

Comment:
Same here. Is having named return values a part of our style guide?

Reply:
For the most part, yes. I believe it saves a stack variable slot, which is pretty valuable in some cases.